crates/oxc_ast/src/trivia.rs

```diff
@@ -126,14 +137,14 @@ impl<'a> CommentsRange<'a> {
 }
 
 impl<'c> Iterator for CommentsRange<'c> {
-    type Item = (&'c u32, &'c Comment);
+    type Item = &'c Comment;
 
     fn next(&mut self) -> Option<Self::Item> {
         if self.current_start < self.current_end {
-            for (start, comment) in &self.comments[self.current_start..self.current_end] {
+            for comment in &self.comments[self.current_start..self.current_end] {
                 self.current_start = self.current_start.saturating_add(1);
-                if self.range.contains(start) {
-                    return Some((start, comment));
+                if self.range.contains(&comment.span.start) {
```

Comment:
I think this can be `comment.span.start` (without `&`).

Reply:
That doesn't seem to build, the trait method takes a reference there: https://doc.rust-lang.org/std/ops/trait.RangeBounds.html#method.contains